src/Patcher/GnuPatchPatcher.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace cweagans\Composer\Patcher;
+
+use Composer\IO\IOInterface;
+use cweagans\Composer\Patch;
+
+class GnuPatchPatcher extends PatcherBase
+{
+    protected string $tool = 'patch';
+
+    public function apply(Patch $patch, string $path): bool
+    {
+        // TODO: Dry run first?
+
+        return $this->executeCommand(
+            'patch -p%s --no-backup-if-mismatch -d %s < %s',
+            $patch->depth,
+            $path,
+            $patch->localPath
+        );
+    }
+
+    public function canUse(): bool
+    {
+        $output = "";
+        $result = $this->executor->execute($this->patchTool() . ' --version', $output);
```

Comment:
I think we also need to test for `gpatch` as patchTool(), or maybe that would just be another Patcher provider (GPatchPatcher)where we just change the value of `$this->tool` ?
`
class GPatchPatcher extends GnuPatchPatcher {

protected string $tool = 'gpatch';

}
`

Reply:
Good idea! I added that.